Gems/Terrain/Assets/Shaders/Terrain/TerrainDetailHelpers.azsli

```diff
@@ -225,26 +250,105 @@ void GetDebugDetailSurface(inout DetailSurface surface, uint material1, uint mat
 {
-    uint4 material1 = TerrainSrg::m_detailMaterialIdImage.GatherRed(TerrainSrg::DetailSampler, idUv, 0).xyzw;
-    uint4 material2 = TerrainSrg::m_detailMaterialIdImage.GatherGreen(TerrainSrg::DetailSampler, idUv, 0).xyzw;
+    GetDetailSurfaceForMaterial(tempSurface, material1, detailUv);
+    if (material2 != 0xFF)
+    {
+        WeightDetailSurface(tempSurface, weight * blend);
+        AddDetailSurface(surface, tempSurface);
+
+        GetDetailSurfaceForMaterial(tempSurface, material2, detailUv);
+        WeightDetailSurface(tempSurface, weight * (1.0 - blend));
+        AddDetailSurface(surface, tempSurface);
+    }
+    else
+    {
+        WeightDetailSurface(tempSurface, weight);
+        AddDetailSurface(surface, tempSurface);
+    }
+}
 
+bool GetDetailSurface(inout DetailSurface surface, float2 detailMaterialIdUv, float2 detailUv)
+{
+    float2 textureSize;
+    TerrainSrg::m_detailMaterialIdImage.GetDimensions(textureSize.x, textureSize.y);
+
+    float2 detailMaterialIdCoord = detailMaterialIdUv * textureSize + 0.5;
+    float2 detailMaterailIdUv = detailMaterialIdCoord;
+    float2 gatherWeight = frac(detailMaterialIdCoord);
+    gatherWeight = 3.0 * gatherWeight * gatherWeight - 2.0 * gatherWeight * gatherWeight * gatherWeight;
+
+    // Using Load() instead of gather because of precision issues with uvs.
+    uint4 s1 = TerrainSrg::m_detailMaterialIdImage.Load(int3((detailMaterialIdCoord + int2(0, 1) + textureSize) % textureSize, 0));
+    uint4 s2 = TerrainSrg::m_detailMaterialIdImage.Load(int3((detailMaterialIdCoord + int2(1, 1) + textureSize) % textureSize, 0));
+    uint4 s3 = TerrainSrg::m_detailMaterialIdImage.Load(int3((detailMaterialIdCoord + int2(1, 0) + textureSize) % textureSize, 0));
+    uint4 s4 = TerrainSrg::m_detailMaterialIdImage.Load(int3((detailMaterialIdCoord + int2(0, 0) + textureSize) % textureSize, 0));
+
+    uint4 material1 = uint4(s1.x, s2.x, s3.x, s4.x);
+    uint4 material2 = uint4(s1.y, s2.y, s3.y, s4.y);
+    
     const float maxBlendAmount = 0xFF;
     // convert integer of 0-255 to float of 0-1.
-    float4 blends = float4(TerrainSrg::m_detailMaterialIdImage.GatherBlue(TerrainSrg::DetailSampler, idUv, 0).xyzw) / maxBlendAmount;
-
+    float4 blends = float4(s1.z, s2.z, s3.z, s4.z) / maxBlendAmount;
+    
     if (o_debugDetailMaterialIds)
     {
+        float2 idUv = (detailMaterialIdCoord + gatherWeight - 0.5) / textureSize;
         GetDebugDetailSurface(surface, material1.x, material2.x, blends.x, idUv);
         return true;
     }
 
-    if (material1.x == 0xFF)
+    if (material1.x == 0xFF || material1.y == 0xFF || material1.z == 0xFF || material1.w == 0xFF)
     {
         return false;
     }
 
-    GetDetailSurfaceForMaterial(surface, material1.x, uv);
+    DetailSurface tempSurface;
+
+    if (material1.x == material1.y && material1.y == material1.z && material1.z == material1.w &&
+        material2.x == material2.y && material2.y == material2.z && material2.z == material2.w)
```

Comment:
Suggestion:  Might be able to reduce to "if (all(material1.x == material1.yzw) && all(material2.x == material2.yzw))" or "if(all(material1 == material1.x) && all(material2 == material2.x))"

Reply:
I can do
```
if (all(material1.x == material1.yzw) && all(material2.x == material2.yzw))
```